src/dpp/streams.cpp

```diff
@@ -0,0 +1,95 @@
+/************************************************************************************
+ *
+ * D++, A Lightweight C++ library for Discord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2021 Craig Edwards and D++ contributors 
+ * (https://github.com/brainboxdotcc/DPP/graphs/contributors)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ************************************************************************************/
+
+
+#include <dpp/streams.h>
+#include <dpp/channel.h>
+
+
+#include <utility>
+
+namespace dpp {
+
+int end_msg() 
+{ 
+	return 0; 
+}
+channel_stream::channel_stream(cluster& bot, const channel& out_channel) {
+	this->bot = &bot;
```

Comment:
Two problems :
1. `bot` is semantically ambiguous here. Please give a different name to the field from the parameter.
2. Can you use constructor initializer lists? This is what we do for new code and what we want to update the library to use where it's missing.
E.g.
```cpp
channel_stream::channel_stream(cluster& bot_, const channel& out_channel) : bot(&bot_), out_channel_id(out_channel.id) {}
```